mesh/nurbs.hpp

```diff
@@ -95,14 +95,16 @@ class NURBSPatch
 
    Array<KnotVector *> kv;
 
-   int sd, nd;
+   int sd, nd, ls;
 
    void swap(NURBSPatch *np);
 
    // Special B-NET access functions
    int SetLoopDirection(int dir);
-   inline       double &operator()(int i, int j);
-   inline const double &operator()(int i, int j) const;
+   inline       double &slice(int i, int j);
```

Comment:
Please add a comment to describe this new function `slice`. It would be helpful to mention how it is different from the `operator()` function, and how it works in different dimensions (1D, 2D, 3D).

Reply:
The slice operator is actually identical to the old operator(). 

I needed to extend the  _Standard B-NET access functions_  as defined in lines 140 - 147 to include the 1D case.
For these direct ijk access to the B_net we use  operator(). 

As mentioned lines 102 - 105 define _Special B-NET access functions_. 
Effectively  the `SetLoopDirection(int dir)` direction flattens the multi-dimensional B-NET to get an effective 1D net that can be modified using 'InsertKnot', 'DegreeElevate' or 'UniformRefinement'. The slice operator is the access function in that flattened structure. The first int gives the slice and the second int the element in that slice.

Long story short:
Nothing changed here, except the name
I will mention this in the comment
I will try to give a short and clear explanation of what slice does...